packages/react-native-bpk-component-text-input/src/BpkTextInput.js

```diff
@@ -185,18 +195,36 @@ class BpkTextInput extends Component<Props, State> {
       valid === false && <InvalidIcon />
     );
 
+    const themeAttributes = getThemeAttributes(
+      REQUIRED_THEME_ATTRIBUTES,
+      theme,
+    );
+
+    const focusedColor = themeAttributes
+      ? themeAttributes.textInputFocusedColor
+      : undefined;
+
     const animatedLabelStyle = getLabelStyle(
       this.animatedValues.color,
       this.animatedValues.labelPosition,
       { value, valid, editable, hasAccessoryView },
+      focusedColor,
     );
 
     const animatedInputStyle = getInputContainerStyle(
       this.animatedValues.color,
       hasAccessoryView,
       valid,
+      focusedColor,
     );
 
+    const inputTextStyle = [styles.input];
+
+    if (themeAttributes) {
+      inputTextStyle.push({ fontFamily: themeAttributes.textFontFamily });
+      animatedLabelStyle.push({ fontFamily: themeAttributes.textFontFamily });
+    }
+
```

Comment:
We can avoid the double check of `themeAttributes` if we move this `if` block upwards and then add the redefinition of `focusedColor` to it, like:

```
let focusedColor = undefined;
if (themeAttributes) {
      inputTextStyle.push({ fontFamily: themeAttributes.textFontFamily });
      animatedLabelStyle.push({ fontFamily: themeAttributes.textFontFamily });
      focusedColor = themeAttributes.textInputFocusedColor;
}
```

The optimisation is minor but it'll help make it clearer if all the theme logic is in the same place.

Reply:
We can't because the function that returns `animatedLabelStyle` depends on `focusedColor`

Reply:
Ahh - never mind then 😄